src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTagger.cs

```diff
@@ -19,56 +23,95 @@ internal sealed class InlineParameterNameHintsTagger : ITagger<IntraTextAdornmen
         {
+            _cache.Clear();
             var spans = e.Span.GetSpans(_buffer);
             foreach (var span in spans)
             {
                 TagsChanged?.Invoke(this, new SnapshotSpanEventArgs(span));
             }
         }
 
+        private TextFormattingRunProperties Format
+        {
+            get
+            {
+                _format ??= _formatMap.GetTextProperties(_keyword);
+
+                return _format;
+            }
+        }
+
         public IEnumerable<ITagSpan<IntraTextAdornmentTag>> GetTags(NormalizedSnapshotSpanCollection spans)
         {
-            if (spans.Count <= 0)
+            if (spans.Count == 0)
             {
                 return Array.Empty<ITagSpan<IntraTextAdornmentTag>>();
             }
 
-            var tagsList = new List<ITagSpan<IntraTextAdornmentTag>>();
-            var dataTags = _tagAggregator.GetTags(spans);
             var snapshot = spans[0].Snapshot;
-            foreach (var tag in dataTags)
+            if (_cache.Count == 0 || snapshot != _cacheSnapshot)
             {
-                // Gets the associated span from the snapshot span and creates the IntraTextAdornmentTag from the data
-                // tags. Only dealing with the dataTagSpans if the count is 1 because we do not see a multi-buffer case
-                // occuring 
-                var dataTagSpans = tag.Span.GetSpans(snapshot);
-                var textTag = tag.Tag;
-                if (dataTagSpans.Count == 1)
+                // Calculate UI elements
+                _cache.Clear();
+                _cacheSnapshot = snapshot;
+                var fullSpan = new SnapshotSpan(snapshot, 0, snapshot.Length);
+                var requestSpans = new NormalizedSnapshotSpanCollection(fullSpan);
+                var dataTags = _tagAggregator.GetTags(requestSpans);
+                foreach (var tag in dataTags)
                 {
-                    var dataTagSpan = dataTagSpans[0];
-                    tagsList.Add(new TagSpan<IntraTextAdornmentTag>(new SnapshotSpan(dataTagSpan.Start, 0), new InlineParameterNameHintsTag(textTag.ParameterName)));
+                    // Gets the associated span from the snapshot span and creates the IntraTextAdornmentTag from the data
+                    // tags. Only dealing with the dataTagSpans if the count is 1 because we do not see a multi-buffer case
+                    // occuring 
+                    var dataTagSpans = tag.Span.GetSpans(snapshot);
+                    var textTag = tag.Tag;
+                    if (dataTagSpans.Count == 1)
+                    {
+                        var dataTagSpan = dataTagSpans[0];
+                        _cache.Add(new TagSpan<IntraTextAdornmentTag>(new SnapshotSpan(dataTagSpan.Start, 0), new InlineParameterNameHintsTag(textTag.ParameterName, _textView.LineHeight, Format)));
+                    }
                 }
             }
 
-            return tagsList;
+            return _cache;
```

Comment:
We aren't filtering this to only the spans that were asked for? That can _sometimes_ cause downstream surprises but if the editor is filtering them out right away then it's not too much of a concern.